From 3cc6731989568df9bf86588f1c4d1926946d3dbd Mon Sep 17 00:00:00 2001
From: Shraddha Chaudhari <Shraddha.Chaudhari@imgtec.com>
Date: Thu, 13 Aug 2015 18:13:01 +0530
Subject: marduk_dts: enable spidev on SPIM0

Marduk has two mikroBUS interface which can have a SPI device
connected to it. SPI device will vary based on the connected
click board. So enable spidev, this will help in probing or basic
testing of the device.

Disable i2c0 as the pins are shared with SPI chipselects.

Change-Id: I8e95ac772948f09348845572f91f50d4fa409700
Signed-off-by: Shraddha Chaudhari <Shraddha.Chaudhari@imgtec.com>
---
 arch/mips/boot/dts/pistachio/pistachio_marduk.dts | 20 ++++++++++++++------
 1 file changed, 14 insertions(+), 6 deletions(-)

diff --git a/arch/mips/boot/dts/pistachio/pistachio_marduk.dts b/arch/mips/boot/dts/pistachio/pistachio_marduk.dts
index b653408..1a1c8a0 100644
--- a/arch/mips/boot/dts/pistachio/pistachio_marduk.dts
+++ b/arch/mips/boot/dts/pistachio/pistachio_marduk.dts
@@ -89,7 +89,8 @@
 	pinctrl-0 = <&spim0_pins>;
 	pinctrl-names = "default";
 
-	cs-gpios = <&gpio1 14 GPIO_ACTIVE_HIGH>, <&gpio0 2 GPIO_ACTIVE_HIGH>;
+	cs-gpios = <&gpio1 14 GPIO_ACTIVE_HIGH>, <&gpio0 2 GPIO_ACTIVE_HIGH>,
+			<&gpio1 12 GPIO_ACTIVE_HIGH>, <&gpio1 13 GPIO_ACTIVE_HIGH>;
 
 	cc2520@0 {
 		compatible = "ti,cc2520";
@@ -114,6 +115,18 @@
 		gpio-controller;
 		#gpio-cells = <2>;
 	};
+
+	spidev@2 {
+		compatible = "linux,spidev";
+		reg = <2>;
+		spi-max-frequency = <2000000>;
+	};
+
+	spidev@3 {
+		compatible = "linux,spidev";
+		reg = <3>;
+		spi-max-frequency = <2000000>;
+	};
 };
 
 &spfi1 {
@@ -197,11 +210,6 @@
 	vref-supply = <&reg_1v8>;
 };
 
-&i2c0 {
-	status = "okay";
-	clock-frequency = <400000>;
-};
-
 &i2c2 {
 	status = "okay";
 	clock-frequency = <400000>;
-- 
1.9.1

